runtime/cudaq/qis/state.cpp

```diff
@@ -111,6 +111,33 @@ extern "C" {
 std::int64_t __nvqpp_cudaq_state_numberOfQubits(state *obj) {
   return obj->get_num_qubits();
 }
+
+state *__nvqpp_cudaq_state_createFromData_fp64(void *data, std::size_t size) {
+  auto d = reinterpret_cast<std::complex<double> *>(data);
+  auto current = std::vector<std::complex<double>>(d, d + size);
+
+  auto *simulator = cudaq::get_simulator();
+  if (simulator->isSinglePrecision()) {
+    auto converted =
+        std::vector<std::complex<float>>(current.begin(), current.end());
```

Comment:
Would this work?
```suggestion
    auto converted =
        std::vector<std::complex<float>>(d, d + size);
```
and move the `current` constructor to after the `if` block

Reply:
Let's try to use "left-handed" declarations in these cases rather than `auto`.
```suggestion
    std::vector<std::complex<float>> converted (current.begin(), current.end());
```

Reply:
Thanks, will do in the PR update.

Reply:
Done, thanks!